### PR TITLE
feat(tests): integrate Vitest and React Testing Library 

### PR DESCRIPTION
**Pull request by hdinia**
### Summary

This pull request enhances the CI pipeline by integrating Vitest for running unit tests and React Testing Library (RTL) with jsdom for DOM testing.

### Changes Included

- **Installed Vitest for unit testing**
  - Added Vitest as a development dependency.
  - Configured test scripts in `package.json` to use Vitest.

- **Installed React Testing Library and jsdom for DOM testing**
  - Added React Testing Library and jsdom as development dependencies.
  - Included a basic test setup to ensure proper integration.

- **Configured test scripts in `package.json`**
  - Added necessary scripts for building, linting, and testing the project.

- **Updated GitHub Actions workflow to include test steps**
  - Modified the `npm-test` job to include steps for running Vitest tests.
  - Ensured the workflow checks out the repository, sets up Node.js, installs dependencies, and runs tests.

### How to Test

1. **Local Testing**
   - Run `npm install` to install the new dependencies.
   - Execute `npm test` to ensure Vitest and RTL run correctly and all tests pass.

2. **CI Testing**
   - Push changes to a branch and verify that the GitHub Actions workflow runs successfully.
   - Confirm that the `npm-test` job completes without errors and that all tests pass.
   
   
image

### Ressources:
[Vitest](https://vitest.dev/guide/)
[React Testing Library (RTL)](https://testing-library.com/docs/react-testing-library/intro)
[Jsdom](https://github.com/jsdom/jsdom)

### CODE REVIEW

**Review by laurent-laporte-pro — COMMENTED** [review]
J'apprécie vraiment cette initiative d'ajouter des tests unitaires sur le frontend 🎉. C'est quelque chose qui, à mon avis, manquait à ce projet. Je vais aussi étudier attentivement les références documentaires que tu as fournies dans la description.

L'exécution de tests unitaires sur le frontend nécessite généralement de simuler les réponses des endpoints d'API. Il faut veiller à structurer ces mocks de manière logique (en général, on utilise l'arborescence dictée par les routes URL). Attention, cela peut vite devenir très chronophage !

**Review by laurent-laporte-pro — DISMISSED** [review]

File: webapp/package-lock.json [hunk 1]
```diff
@@ -87,10 +90,13 @@
         "@types/react-window": "1.8.6",
         "@types/redux-logger": "3.0.10",
         "@types/swagger-ui-react": "4.18.1",
+        "@types/testing-library__jest-dom": "6.0.0",
         "@types/uuid": "9.0.4",
         "@typescript-eslint/eslint-plugin": "6.14.0",
         "@typescript-eslint/parser": "6.14.0",
         "@vitejs/plugin-react-swc": "3.5.0",
+        "@vitest/coverage-v8": "^1.6.0",
```
Comment (skamril) [inline webapp/package-lock.json]: Remove all `^`

File: webapp/vite.config.ts [hunk 2]
```diff
@@ -1,3 +1,4 @@
+/// <reference types="vitest" />
```
Comment (skamril) [inline webapp/vite.config.ts:1]: Why this line?

File: webapp/package-lock.json (hunk 1 above)
Comment (hdinia) [reply webapp/package-lock.json]: its not recommended to manually edit `package-lock.json`, I will try re-run npm install

File: webapp/vite.config.ts (hunk 2 above)
Comment (hdinia) [reply webapp/vite.config.ts:1]: reference for vitest types, you can read this: https://www.typescriptlang.org/docs/handbook/triple-slash-directives.html#-reference-types-

**Comment by hdinia** [comment]
Added a simple test to allow the `npm test` check, because without tests it does not pass.

**Review by laurent-laporte-pro — APPROVED** [review]

File: webapp/package-lock.json (hunk 1 above)
Comment (skamril) [reply webapp/package-lock.json]: Yes I comment the wrong file

**Pull request merged**